scripts/build.ts

```diff
@@ -0,0 +1,22 @@
+import { getGraph, INode } from './Graph';
+import { logSuccessBold, runProcess } from '../packages/utils/src';
+import * as path from 'path';
+import { ORGANISATION } from './Packages';
+
+const run = async () => {
+  const graph = getGraph();
+
+  const paths = [];
+  graph.walk((node: INode) => {
+    paths.push(path.join(path.resolve(__dirname, '..'), 'packages', node.name).replace(ORGANISATION + '/', ''));
+  });
+
+  for (const packagePath of paths) {
+    await runProcess('lerna', ['bootstrap', '--force-local'], { silent: true });
+    await runProcess('npm', ['run', 'build'], { cwd: packagePath, silent: true });
```

Comment:
To execute `npm run build` in all the packages, we use the `getGraph` and loop over the results to make `paths` array.
Instead of that we can use lerna run command which will run a npm script for all the packages. The time difference is huge at least in my laptop. 

so in run function all we can write is

`await runProcess('lerna', ['bootstrap', '--force-local'], { silent: true });`
`await runProcess('lerna', ['run', '--stream', 'build']);`

It also takes care of logging output for the packages which were build successfully

Reply:
With lerna run it took 20 seconds and with npm run build manually 150 - 180 seconds

Reply:
Actually just noticed it's not npm run build but it's lerna bootstrap which takes time. But I think lerna bootstrap should be executed only once.

Reply:
Let's talk about this one. I introduced the dependency graph because it's not possible to run `lerna run build` if you check out the repo for the first time. E.G. the service throws an error because the utils package is not already built. Same goes for the dev command.

And If you do the same thing with only running `lerna bootstrap` in the beginning, it won't work either. I don't know why but I had to add it to the build script.

I had a look at other lerna repos and it seems an unsolved problem. Same goes for updating the lerna version.

Reply:
can you test `lerna boostrap` outside the for of loop. I just tried by deleting all the `node_modules` and `package-lock.json` and moved `lerna boostrap` command outside for of loop to execute only once and build was successful.

Reply:
Also regarding `lerna run build`, it takes care of the dependency graph. For reference -- https://github.com/lerna/lerna/issues/1681. And I also tested on my local. If we pass parallel flag, then we will face the issue you mentioned.

Reply:
How I tested it:

- `lerna run clean`
- `lerna clean`
- `lerna bootstrap`
- `lerna run build`

And it works. That's pretty cool, I tried it last week and it didn't work. I have to verify it on my other machine.

If this is gonna work now, I will remove the build script.

But I think the dev and release scripts are still valid, what do you think?

Reply:
For dev also the script can be removed and `lerna run --parallel dev` command can be used. Also it will output message when was the last changed package.

Reply:
I tried it with `build` on my machine and it works but `dev` doesn't work because since it runs in parallel it doesn't have the right order:

```
@vuesion/service: src/generators/connected/index.ts(2,31): error TS2307: Cannot find module '@vuesion/models'.
@vuesion/service: src/generators/connected/index.ts(3,29): error TS2307: Cannot find module '@vuesion/utils/dist/path'.
@vuesion/service: src/generators/connected/index.ts(4,30): error TS2307: Cannot find module '@vuesion/utils/dist/fileSystem'.
@vuesion/service: src/generators/module/index.ts(3,31): error TS2307: Cannot find module '@vuesion/models'.
@vuesion/service: src/generators/module/index.ts(4,29): error TS2307: Cannot find module '@vuesion/utils/dist/path'.
@vuesion/service: src/generators/module/index.ts(5,30): error TS2307: Cannot find module '@vuesion/utils/dist/fileSystem'.
@vuesion/service: src/index.ts(3,30): error TS2307: Cannot find module '@vuesion/utils/dist/path'.
@vuesion/service: src/index.ts(4,26): error TS2307: Cannot find module '@vuesion/models'.
@vuesion/service: src/index.ts(5,32): error TS2307: Cannot find module '@vuesion/models'.
@vuesion/service: src/index.ts(6,32): error TS2307: Cannot find module '@vuesion/models'.
@vuesion/service: src/scripts/extract-i18n-messages.ts(5,31): error TS2307: Cannot find module '@vuesion/models'.
@vuesion/service: src/scripts/extract-i18n-messages.ts(6,39): error TS2307: Cannot find module '@vuesion/utils/dist/ui'.
@vuesion/service: src/scripts/extract-i18n-messages.ts(7,39): error TS2307: Cannot find module '@vuesion/utils/dist/fileSystem'.
@vuesion/service: src/scripts/run-webpack.ts(2,43): error TS2307: Cannot find module '@vuesion/utils/dist/ui'.
@vuesion/service: src/scripts/update.ts(4,29): error TS2307: Cannot find module '@vuesion/utils/dist/path'.
@vuesion/service: src/scripts/update.ts(5,39): error TS2307: Cannot find module '@vuesion/utils/dist/fileSystem'.
@vuesion/service: src/scripts/update.ts(6,78): error TS2307: Cannot find module '@vuesion/utils/dist/ui'.
@vuesion/service: src/scripts/update.ts(7,31): error TS2307: Cannot find module '@vuesion/models'.
@vuesion/service: 8:59:45 PM - Found 98 errors. Watching for file changes.
```

we could add a `predev` script which builds all packages before the dev mode starts. What do you think?

Reply:
Yeah.. That's a nice workaround.